p2p/host/autorelay/relay_finder.go

```diff
@@ -232,10 +331,17 @@ func (rf *relayFinder) handleNewNode(ctx context.Context, pi peer.AddrInfo) {
 		return
 	}
 	log.Debugw("node supports relay protocol", "peer", pi.ID, "supports circuit v2", supportsV2)
-	rf.candidates[pi.ID] = &candidate{ai: pi, supportsRelayV2: supportsV2}
+	rf.candidates[pi.ID] = &candidate{
+		added:           rf.conf.clock.Now(),
+		ai:              pi,
+		supportsRelayV2: supportsV2,
+	}
 	rf.candidateMx.Unlock()
 
-	rf.notifyNewCandidate()
+	// Don't notify when we're using static relays. We need to process all entries first.
+	if !rf.usesStaticRelay() {
+		rf.notifyNewCandidate()
```

Comment:
It's not ideal that this function has a notion of "static relay setup".

I think it would be better if this function returned whether this node was added as a candidate, and the caller should then decide if it wants to notify new candidate. This way the `handleStaticRelays` caller can wait until the end, and the findNodes caller can notifyNewCandidate if it was added.

Reply:
Good suggestion! Applied in f5ea2ef58487bed41c6e94cebf8ea18423189110.